### ISSUE TITLE
Integration cannot be loaded after 2022.12 update

### ISSUE THREAD

**Issue by figorr** [barleybobs/homeassistant-ecowater-softener#26]
Hi,

After upgrading to 2022.12 seems something is broken. The integration cannot be loaded after restarting HA and I got the following log errors.

```
Logger: homeassistant.setup
Source: setup.py:202 
First occurred: 08:37:29 (1 occurrences) 
Last logged: 08:37:29

Setup failed for custom integration ecowater_softener: Requirements for ecowater_softener not found: ['ecowater-softener==1.0.0'].
``` 

```
Logger: homeassistant.util.package
Source: util/package.py:98 
First occurred: 08:37:29 (1 occurrences) 
Last logged: 08:37:29

Unable to install package ecowater-softener==1.0.0: WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f1be795ba30>: Failed to establish a new connection: [Errno -3] Try again')': /musllinux/ 
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f1be7958130>: Failed to establish a new connection: [Errno -3] Try again')': /musllinux/ 
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f1be7959a80>: Failed to establish a new connection: [Errno -3] Try again')': /musllinux/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f1be79dd2a0>: Failed to establish a new connection: [Errno -3] Try again')': /musllinux/ 
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f1be7901840>: Failed to establish a new connection: [Errno -3] Try again')': /musllinux/ 
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f1be77d1300>: Failed to establish a new connection: [Errno -3] Try again')': /simple/ecowater-softener/ 
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f1be7b23b20>: Failed to establish a new connection: [Errno -3] Try again')': /simple/ecowater-softener/ 
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f1be7b23c10>: Failed to establish a new connection: [Errno -3] Try again')': /simple/ecowater-softener/ 
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f1be7b22fe0>: Failed to establish a new connection: [Errno -3] Try again')': /simple/ecowater-softener/ 
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f1be7b23fa0>: Failed to establish a new connection: [Errno -3] Try again')': /simple/ecowater-softener/ 
ERROR: Could not find a version that satisfies the requirement ecowater-softener==1.0.0 (from versions: none) ERROR: No matching distribution found for ecowater-softener==1.0.0 WARNING: There was an error checking the latest version of pip.
``` 

I tried to redownloading the repository from HACS (main and the 2.0.0 too) and restarting HA … with no luck. The integration cannot be loaded after the restart.

If I delete the integration and I try to reinstall it … there is no success. It cannot be installed. Error 500

What it is weird is that the error is referring to version 1.0.0 when I already had updated it to 2.0.0  a long time ago.

**Comment by figorr** [comment]
One curios thing ... I installed a new Home Assistant and I made the setup from the beginning (not using backup). I installed HACS and the first integration I downloaded fromg HACS was the ECOWATER Softener ... I restarted and when I tried to install the integration from the "Integrations" section ... it was not possible.

I got error 500. "Internal Server Error"

![image](https://user-images.githubusercontent.com/77553975/206781688-90c66cc3-a172-4a88-8c07-9b81255da91c.png)

**Comment by figorr** [comment]
Finally I move the NAS to a new location … I connected it to a different network and … homeassistant was able to load the Ecowater Softener integration without any problem… so the files that the integration needs (and that were deleted or corrupted after installing homeassistant 2022.12) were downloaded and installed correctly.

Now I moved again the NAS to home … and the integration was loaded Ok.

So maybe there is a problem with the router … that was blocking something (although I didn’t change anything recently) o maybe it was a problem with the ISP.

I am going to close this thread.
